scipy/stats/_qmc.py

```diff
@@ -1036,6 +1081,49 @@ def _random(self, n: IntNumber = 1) -> np.ndarray:
         self.num_generated += n
         return samples
 
+    def _random_oa_lhs(self, n: IntNumber = 1) -> np.ndarray:
+        """Orthogonal array based LHS of strength 2."""
+        n_row = self._p**2
+        n_col = self._p + 1
+
+        if n != n_row:
+            raise ValueError(
+                "n is not equal to orthogonal_array_p**2"
+            )
+
+        oa_sample = np.zeros(shape=(n_row, n_col), dtype=int)
+
+        # OA of strength 2
+        arrays = np.tile(np.arange(self._p), (2, 1))
+        oa_sample[:, :2] = np.stack(np.meshgrid(*arrays),
+                                    axis=-1).reshape(-1, 2)
+        for p_ in range(1, self._p):
+            oa_sample[:, 2+p_-1] = np.mod(oa_sample[:, 0]
+                                          + p_*oa_sample[:, 1], self._p)
+
+        # scramble the OA
+        oa_sample_ = np.empty(shape=(n_row, n_col), dtype=int)
+        for j in range(n_col):
+            perms = self.rng.permutation(self._p)
+            oa_sample_[:, j] = perms[oa_sample[:, j]]
+
+        # following is making a scrambled OA into an OA-LHS
+        oa_lhs_sample = np.zeros(shape=(n_row, n_col))
+        lhs_engine = LatinHypercube(d=1, centered=self.centered, seed=self.rng)
+        for j in range(n_col):
+            for k in range(self._p):
+                idx = np.where(oa_sample[:, j] == k)[0]
+                lhs = lhs_engine.random(self._p).flatten()
+                oa_lhs_sample[:, j][idx] = lhs + oa_sample[:, j][idx]
+
+                lhs_engine = lhs_engine.reset()
+
+        oa_lhs_sample /= self._p
+
+        idx = self.rng.permutation(n_col)[:self.d]
+        oa_lhs_sample = oa_lhs_sample[:, idx]
```

Comment:
I am not sure if you need this after you applied random permutation of symbols for each column independently... As far as I can remember, Owen, A. B. (1992) did not apply any permutation of columns.

Reply:
Agreed I will remove this.